src/telemetry.rs

```diff
@@ -84,6 +87,55 @@ struct TestResponse {
     pub test_id: String,
 }
 
+fn redact_string(_val: &str) -> String {
+    static REDACTED_VALUE: &str = "*********";
+    REDACTED_VALUE.to_string()
+}
+
+fn redact_headers(headers: &mut Vec<Header>) {
+    let should_redact = |s: &str| {
+        return s.to_lowercase() == "authorization";
+    };
+
+    headers
+        .iter_mut()
+        .filter(|h| should_redact(h.header.as_str()))
+        .for_each(|h| {
+            return h.value = redact_string(&h.value);
+        });
+}
+
+fn redact_request(req: &mut RequestDescriptor) {
+    redact_headers(req.headers.as_mut())
+}
+
+fn redact_definition(mut td: test::Definition) -> test::Definition {
+    _ = td.setup.as_mut().map(|s| redact_request(&mut s.request));
+
+    _ = td.cleanup.always.as_mut().map(redact_request);
```

Comment:
Is there a benefit to using `.map()` here as opposed to just passing `td.cleanup.always.as_mut()` into the `redact_request()` function?

Reply:
If I'm following your question correctly, always is an `Option<RequestDescriptor>` and the `redact_request` function accepts `RequestDescriptor` types. So I lift the function using `map` .

Reply:
Sadly rust does not have an ifPresent 😄